src/fetchtastic/download/client_app.py

```
@@ -1220,6 +1364,423 @@ def check_extraction_needed(
+                from datetime import datetime
+
+                dt = datetime.fromisoformat(published_at.replace("Z", "+00:00"))
+                dir_name = f"{dt.strftime('%Y%m%d-%H%M%S')}-{version_code}"
+            except (ValueError, TypeError):
+                pass
+        release_dir = os.path.join(
+            base_dir,
+            self._sanitize_required(dir_name, "snapshot directory name"),
+        )
+        if os.path.islink(release_dir):
+            raise ValueError(f"Refusing symlinked snapshot release dir: {release_dir}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)